artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTSessionCallback.java

```diff
@@ -36,6 +36,11 @@ public MQTTSessionCallback(MQTTSession session, MQTTConnection connection) throw
       this.connection = connection;
    }
 
+   @Override
+   public boolean supportsDirectDelivery() {
```

Comment:
if its a deadlock issue in queue, this shouldn't be needed. We should avoid using a sledge hammer by disabling features. fix the deadlock not just disable stuff.

Reply:
According to @clebertsuconic is not that way: there are protocols that doesn't support direct deliver and I seems that mqtt is one of them.

Reply:
Still the point is you're fixing a deadlock in this PR, just fix the deadlock, don't make other changes. If we have deadlock in QueueImpl, then its irregardless of these other changes.

Reply:
I could split the pr in several ones but they are tightly related, but yes, can be done for sure

Reply:
I don't think MQTT was supposed to have direct delivery support. There's conversion on messages.

Reply:
@clebertsuconic i don't see why that impacts if it can direct deliver or not, direct deliver in my understanding is if it can use the producer thread, to avoid needing to context switch for perf, as long as transform is non blocking

I iterate though this bug if i understand the change and franz, is theres deadlock in queueimpl. NOT the protocols.